test/conformance/envvars_test.go

```diff
@@ -0,0 +1,77 @@
+// +build e2e
+
+/*
+Copyright 2018 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package conformance
+
+import (
+	"encoding/json"
+	"strconv"
+	"testing"
+
+	"github.com/knative/pkg/test/logging"
+	"github.com/knative/serving/test"
+)
+
+//TestShouldEnvVars verifies environment variables that are declared as "SHOULD be set" in runtime-contract
+func TestShouldEnvVars(t *testing.T) {
```

Comment:
from code, I see that we validate Must and Should in a very similar way, i.e. check whether they are specified. I could be missing something.  Can you call out the validation difference in comments so that it's obvious?

Reply:
The reason for breaking them into two tests is to adhere to the semantics in the runtime contract as they might have different policies (like SHOULD generating warnings and MUST generating failures). As this is still to be decided, the PR currently just breaks them into two tests and fails on both instances.